modules/cats/test/src/ShowVisitorSpec.scala

```diff
@@ -0,0 +1,258 @@
+package smithy4s.interopcats
+
+import smithy4s.{ByteArray, Hints, ShapeId, Timestamp}
+import smithy4s.interopcats.SchemaVisitorShow
+import smithy4s.schema.Schema
+import smithy4s.schema.Schema._
+import weaver.FunSuite
+import smithy4s.interopcats.testcases.FooBar
+
+import smithy4s.interopcats.testcases.IntOrString.schema
+import smithy4s.interopcats.testcases.IntOrString._
+
+object ShowVisitorSpec extends FunSuite {
+
+  val schemaVisitorShow = SchemaVisitorShow
+
+  test("int") {
+    val schema: Schema[Int] = int
+    val intValue = 1
+    val showOutput = schemaVisitorShow(schema).show(intValue)
+    expect.eql(showOutput, "1")
+  }
+
+  test("string") {
+    val schema: Schema[String] = string
+    val foo = "foo"
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "foo")
+  }
+
+  test("boolean") {
+    val schema: Schema[Boolean] = boolean
+    val foo = true
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "true")
+  }
+
+  test("long") {
+    val schema: Schema[Long] = long
+    val foo = 1L
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1")
+  }
+
+  test("short") {
+    val schema: Schema[Short] = short
+    val foo = 1.toShort
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1")
+  }
+
+  test("byte") {
+    val schema: Schema[Byte] = byte
+    val foo = 1.toByte
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1")
+
+  }
+
+  test("double") {
+    val schema: Schema[Double] = double
+    val foo = 1.0
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1.0")
+
+  }
+
+  test("float") {
+    val schema: Schema[Float] = float
+    val foo = 1.0f
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1.0")
+  }
+
+  test("bigint") {
+    val schema: Schema[BigInt] = bigint
+    val foo = BigInt(1)
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1")
+  }
+
+  test("bigdecimal") {
+    val schema: Schema[BigDecimal] = bigdecimal
+    val foo = BigDecimal(1)
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, "1")
+
+  }
+
+  test("smithy4s ByteArray") {
+    val schema: Schema[ByteArray] = bytes
+    val fooBar = ByteArray("fooBar".getBytes)
+    val showOutput = schemaVisitorShow(schema).show(fooBar)
+    expect.eql(showOutput, "Zm9vQmFy")
+  }
+
+  test("smithy4s timestamp") {
+    val schema: Schema[Timestamp] = timestamp
+    val now = java.time.Instant.now()
+    val foo = Timestamp.fromEpochSecond(now.getEpochSecond)
+    val showOutput = schemaVisitorShow(schema).show(foo)
+    expect.eql(showOutput, foo.toString)
+  }
+
+  test("struct") {
+    case class Foo(x: String, y: Option[String])
+    object Foo {
+      val schema: Schema[Foo] = {
+        StructSchema(
```

Comment:
Use `Schema.struct` to construct structure schemas, it'll give you type-safe methods for arity 1 to 22 that do not require casting.